[Zhang Yuxiang] iP

TheRareFox opened the pull request:
# Danny Bot
This cute ~~human slave~~ cat bot will always be here for you to help you record your tasks!
> . Nice cat :D - Darren ([Source](https://github.com/nus-cs2103-AY2425S1/ip/pull/487))

This cat is 
- CLI based
- Cute :wink:
- Creative
- Cunning

It can
1. Create Todos `todo [name]`
2. Create Deadlines `deadline [name] /by [date]`
3. Create Events `event [name] /from [date] /to [date]`
4. Keep track of the tasks `list`
5. Save and load tasks when quitting
6. And more...

To clone this code and run it on your local machine:
```
git clone https://github.com/TheRareFox/ip.git
```

Is this good?
- [ ] Yes
- [x] Even better than good

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TheRareFox committed:
Changed text bot to Danny

TheRareFox committed:
Updated Chatbot with echoing functionalities and ability to say bye

TheRareFox committed:
- Added "add" command to add items to list
- Added "list" command to display list

TheRareFox committed:
- Added mark feature to list
- Added unmark feature to list
- Added elusive cat to cat-ch errors

TheRareFox committed:
Added OOP functionality to Task

TheRareFox committed:
Added ToDos, Deadlines and Events

TheRareFox committed:
Added Error Handling and throwing exceptions

TheRareFox committed:
Added Delete Functionality for the list

TheRareFox committed:
Added Enum

TheRareFox committed:
Refactor code to make it more structured

TheRareFox committed:
Add Saving and Loading Functionality

TheRareFox committed:
Merge branch 'branch-level-7'

TheRareFox committed:
Add Datetime to Tasks

TheRareFox committed:
Refactor code to make it adhere to more OOP principles

TheRareFox committed:
Refactor to add more packages

TheRareFox committed:
Merge branch 'add-gradle-support'

TheRareFox committed:
Add Junit tests on Parser

TheRareFox committed:
Add new file capability

TheRareFox committed:
Add JavaDocs

TheRareFox committed:
Add Find for tasks

TheRareFox committed:
Merge branch 'branch-A-JavaDoc'

TheRareFox committed:
Refactor code

Darren-Tung commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,38 @@
+        List<Task> savedTasks = new ArrayList<>();
+        Storage loader = null;
+        try {
+            loader = new Storage("src/main/java/data/tasks.txt",danny);
```
You can consider adding a space after the comma and before 'danny'

Darren-Tung commented on src/main/java/impl/chatbot/Danny.java:
```diff
@@ -0,0 +1,41 @@
+package impl.chatbot;
```
You can consider a better name for your 'impl' file.

Darren-Tung commented on src/main/java/impl/ui/Parser.java:
```diff
@@ -0,0 +1,221 @@
+
+
+    public static Command checkCommand(String in, int size) throws InvalidCommandException{
+        String test = in.toLowerCase().split(" ")[0];
```
You may consider using another name than 'test'

Darren-Tung commented on src/main/java/impl/ui/Ui.java:
```diff
@@ -0,0 +1,38 @@
+package impl.ui;
+
+public class Ui {
+    static String cat = """
```
Nice cat :D

Darren-Tung reviewed (commented):
Overall well done! Only minor changes to look at.

tyxiangs commented on src/main/java/impl/ui/Parser.java:
```diff
@@ -0,0 +1,221 @@
+            cmd = Parser.Command.Error;
+        }
+
+        switch (cmd){
+            case List:
+                handleList();
+                break;
+            case Todo:
+                handleToDo(in);
+                break;
+            case Deadline:
+                handleDeadline(in);
+                break;
+            case Event:
+                handleEvent(in);
+                break;
+            case Mark:
+                handleMark(in);
+                break;
+            case Unmark:
+                handleUnmark(in);
+                break;
+            case Delete:
+                handleDelete(in);
+                break;
+            case Find:
+                handleFind(in);
+                break;
+            case Error:
+                System.out.println("Error! Please try again :(");
+                break;
+            case Unknown:
+                // Fall through as default
+            default:
+                System.out.println("Unknown Command Detected!");
+                break;
```
I think you might need to modify the style for switch statements as
   switch() { 
   case "T" : 
'case' must be aligned with 'switch'
![image](https://github.com/user-attachments/assets/6047da43-06b7-4e59-9c8f-f04f5c5b85c5)

tyxiangs commented on src/main/java/impl/ui/Parser.java:
```diff
@@ -0,0 +1,221 @@
+        String find = in.split(" ",2)[1];
+        String text;
+        for (int i = 0; i < list.size(); i++) {
+            if(list.get(i).getDescription().toLowerCase().contains(find.toLowerCase())){
```
I think this line might be too long, you may consider extending it to the next line.

tyxiangs commented on src/main/java/impl/ui/Parser.java:
```diff
@@ -0,0 +1,221 @@
+package impl.ui;
+
+import exceptions.InvalidCommandException;
+import impl.storage.TaskList;
+import impl.interfaces.Deadlines;
+import impl.interfaces.Events;
+import impl.interfaces.Task;
+import impl.interfaces.ToDos;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
```
I think you need to put all the import java.* statement above the import.other.* statement

tyxiangs commented on src/main/java/impl/storage/Storage.java:
```diff
@@ -0,0 +1,51 @@
+package impl.storage;
+
+import impl.chatbot.Danny;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.Scanner;
```
I think you need to put all the import java.* statement above all the import other.* statement

tyxiangs commented on src/main/java/impl/chatbot/Danny.java:
```diff
@@ -0,0 +1,41 @@
+package impl.chatbot;
+
+import impl.storage.TaskList;
+import impl.ui.Parser;
+import impl.ui.Ui;
+
+import java.util.Scanner;
```
I think you need to put all the import java.* statement above import other.*
![Screenshot 2024-09-05 165433](https://github.com/user-attachments/assets/1c4e7662-8d6e-431b-8694-40367b0bf072)

tyxiangs commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,38 @@
+import impl.chatbot.Danny;
+import impl.storage.Storage;
+import impl.interfaces.Task;
+
+import java.util.ArrayList;
+import java.util.List;
```
I think you need to put all the import java.* statement above the import other.* statement

tyxiangs reviewed (commented):
Overall, a really good job done 👍
I just highlighted a few minor changes which could be reviewed.

TheRareFox committed:
Add GUI functionality

TheRareFox committed:
Add GUI functionality

TheRareFox committed:
Assert statements: Make assert statement

Assert statements should exist to document important assumptions that should hold.

Added assert statements in places that receives inputs to ensure input string is not empty.

TheRareFox committed:
Refactor code for neatness

TheRareFox committed:
Add style

TheRareFox committed:
Merge pull request #3 from TheRareFox/branch-A-CodeQuality

Refactor code for neatness

TheRareFox committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/gui/MainWindow.java
#	src/main/java/impl/chatbot/Danny.java

TheRareFox committed:
Merge pull request #2 from TheRareFox/branch-A-Assertions

Assert statements: Make assert statement

TheRareFox committed:
Add tagging to list

TheRareFox committed:
Add better GUI

TheRareFox committed:
Fix tags not saving

TheRareFox committed:
Update README.md and Javadocs

TheRareFox committed:
Made code neater

TheRareFox committed:
Update README.md

TheRareFox committed:
Update README.md

TheRareFox committed:
Update saving

TheRareFox committed:
Merge remote-tracking branch 'origin/master'

TheRareFox committed:
Update README.md
